lib/dockerregistry/inventory.go

```diff
@@ -688,6 +701,34 @@ func (sc *SyncContext) ReadRepository(
 	sc.ExecRequests(populateRequests, processRequest)
 }
 
+// MkReadRepositoryCmdReal creates a stream.Producer which makes a real call
+// over the network.
+func MkReadRepositoryCmdReal(
+	sc *SyncContext,
+	rc RegistryContext) stream.Producer {
+
+	var sh stream.HTTP
+
+	tokenKey, domain, repoPath := GetTokenKeyDomainRepoPath(rc.Name)
+
+	httpReq, err := http.NewRequest(
```

Comment:
Non-blocker nit. Do we want to add some sort of timeout / retry for this.

Reply:
Yes, but if we're going to add it here we should add it everywhere. But even so, I don't think there's a real need to have this be robust over network failures. This is because we can simply re-run the promoter when the network recovers.

Reply:
cool